### PR TITLE
feat: make li.fi great again

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

The recent issues related to Li.Fi swaps failing were related to an additional "fee collection step", which was spotted while testing against Jumper, which does *not* contain this step. See:

image

The reason is small enough trades which do not have a fee (or any trade which has a 0 fee really) would trigger this step on our side, but not in Jumper, resulting in different calldata, and Li.Fi endpoint giving us some calldata which would eventually revert in the world of lovely easy to debug EVM state machines. This seems to be a recent change on the Li.Fi side, as things were previously working for us there.

Fixed by passing an explicit undefined instead of 0 for the fee step when the affiliate fee is 0, which results in Li.Fi quote not yielding a fee collection anymore, and returning us some valid calldata.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/6485

Low - can't break what's broken, though this re-enables Li.Fi so ensure this *does* work obviously

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Li.Fi token swaps are happy end-to-end and do not fail at our estimation step (`estimate`) network request
- Li.Fi native assets are happy end-to-end and do not fail at our estimation step (`estimate`) network request

- Test the above both on trades with fees and without fees

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

- Li.Fi swaps previously failing are now happy (smolish trade without fee)

Screenshot 2024-03-22 at 14 57 44

- Li.Fi swaps with affiliate fees are happy

Obviously couldn't sign this guy because of frame-injected wallet, but confirmed estimation and Tx building is happy again

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6543?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6543** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6543?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6542** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6542?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gomesalexandre and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/lib/swapper/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts [hunk 1]
```diff
@@ -88,7 +89,9 @@ export async function getTradeQuote(
       slippage: Number(slippageTolerancePercentageDecimal),
       bridges: { deny: ['stargate', 'amarok', 'arbitrum'] },
       allowSwitchChain: true,
-      fee: convertBasisPointsToDecimalPercentage(affiliateBps).toNumber(),
+      fee: affiliateBpsDecimalPercentage.isZero()
+        ? undefined
+        : affiliateBpsDecimalPercentage.toNumber(),
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts:94]: Yes, that line is what made Li.Fi borked.

File: .env.app [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # feature flags
 REACT_APP_FEATURE_CHATWOOT=true
-REACT_APP_FEATURE_LIFI_SWAP=false
+REACT_APP_FEATURE_LIFI_SWAP=true
```
Comment (kaladinlight) [inline .env.app:3]:
```suggestion
```

File: .env.private [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # feature flags
 REACT_APP_FEATURE_MIXPANEL=false
 REACT_APP_FEATURE_CHATWOOT=false
-REACT_APP_FEATURE_LIFI_SWAP=false
+REACT_APP_FEATURE_LIFI_SWAP=true
```
Comment (kaladinlight) [inline .env.private:4]:
```suggestion
```

File: .env.app (hunk 2 above)
Comment (kaladinlight) [reply .env.app:3]: already set in base env

**Review by kaladinlight — APPROVED** [review]

File: .env.app (hunk 2 above)
Comment (gomesalexandre) [reply .env.app:3]: Might as well keep it ? This was a revert of https://github.com/shapeshift/web/pull/6524 which disabled it on prod envs only.

File: .env.private (hunk 3 above)
Comment (gomesalexandre) [reply .env.private:4]: Might as well keep it ? This was a revert of https://github.com/shapeshift/web/pull/6524 which disabled it on prod envs only.

**Pull request merged**